opendbc/car/__init__.py

```diff
@@ -100,6 +114,41 @@ class Bus(StrEnum):
   ap_party = auto()
 
 
+def get_max_angle_delta(v_ego_raw: float, steer_step: int, VM: VehicleModel, max_lateral_jerk: float = MAX_LATERAL_JERK):
+  """ Calculate the maximum steering angle rate based on lateral jerk limits. """
+  max_curvature_rate_sec = max_lateral_jerk / (v_ego_raw ** 2)  # (1/m)/s
+  max_angle_rate_sec = math.degrees(VM.get_steer_from_curvature(max_curvature_rate_sec, v_ego_raw, 0))  # deg/s
+  return max_angle_rate_sec * (DT_CTRL * steer_step)
+
+
+def get_max_angle(v_ego_raw: float, VM: VehicleModel, max_lateral_accel: float = MAX_LATERAL_ACCEL):
+  """ Calculate the maximum steering angle based on lateral acceleration limits. """
+  max_curvature = max_lateral_accel / (v_ego_raw ** 2)  # 1/m
+  return math.degrees(VM.get_steer_from_curvature(max_curvature, v_ego_raw, 0))  # deg
+
+
+def apply_common_steer_angle_limits(apply_angle: float, apply_angle_last: float, v_ego_raw: float, steering_angle: float, lat_active: bool,
+                                    limits: AngleSteeringLimits, steer_step: int, VM: VehicleModel) -> float:
```

Comment:
make sure you clearly distinguish these angle functions from the existing breakpoint-based function that will be deprecated in the future. it's not clear they're separate

Reply:
going for `apply_vm_steer_angle_limits`

Reply:
Done

Reply:
yeah just copy what safety.h calls it

Reply:
You don't have these methods defined there  you do it differently so not 1:1 

```cpp
const float max_curvature_rate_sec = MAX_LATERAL_JERK / (fudged_speed * fudged_speed);
const float max_angle_rate_sec = get_angle_from_curvature(max_curvature_rate_sec, curvature_factor, params);

// finally get max angle delta per frame
const float max_angle_delta = max_angle_rate_sec / (float)limits.frequency;
```

Are we okay with the current names I did? 
* `apply_vm_steer_angle_limits`
* `get_max_angle_delta_vm`
* `get_max_angle_vm`